core/src/main/java/com/graphhopper/util/GHUtility.java

```diff
@@ -317,32 +326,55 @@ public static Graph shuffle(Graph g, Graph sortedGraph) {
      */
     public static Graph sortDFS(Graph g, Graph sortedGraph) {
         int nodes = g.getNodes();
-        final GHIntArrayList list = new GHIntArrayList(nodes);
-        list.fill(nodes, -1);
-        final GHBitSetImpl bitset = new GHBitSetImpl(nodes);
-        final AtomicInteger ref = new AtomicInteger(-1);
+        final GHIntArrayList nodeList = new GHIntArrayList(nodes);
+        nodeList.fill(nodes, -1);
+        final GHBitSetImpl nodeBitset = new GHBitSetImpl(nodes);
+        final AtomicInteger nodeRef = new AtomicInteger(-1);
+
+        int edges = g.getEdges();
+        final GHIntArrayList edgeList = new GHIntArrayList(edges);
+        edgeList.fill(edges, -1);
+        final GHBitSetImpl edgeBitset = new GHBitSetImpl(edges);
+        final AtomicInteger edgeRef = new AtomicInteger(-1);
+
         EdgeExplorer explorer = g.createEdgeExplorer();
         for (int startNode = 0; startNode >= 0 && startNode < nodes;
-             startNode = bitset.nextClear(startNode + 1)) {
+             startNode = nodeBitset.nextClear(startNode + 1)) {
             new DepthFirstSearch() {
                 @Override
                 protected GHBitSet createBitSet() {
-                    return bitset;
+                    return nodeBitset;
+                }
+
+                @Override
+                protected boolean checkAdjacent(EdgeIteratorState edge) {
+                    int edgeId = edge.getEdge();
+                    if (!edgeBitset.contains(edgeId)) {
+                        edgeBitset.add(edgeId);
+                        edgeList.set(edgeRef.incrementAndGet(), edgeId);
+                    }
+                    return super.checkAdjacent(edge);
                 }
 
                 @Override
                 protected boolean goFurther(int nodeId) {
-                    list.set(nodeId, ref.incrementAndGet());
+                    nodeList.set(nodeId, nodeRef.incrementAndGet());
                     return super.goFurther(nodeId);
                 }
             }.start(explorer, startNode);
         }
-        return createSortedGraph(g, sortedGraph, list);
+        return createSortedGraph(g, sortedGraph, nodeList, edgeList);
     }
 
-    static Graph createSortedGraph(Graph fromGraph, Graph toSortedGraph, final IntIndexedContainer oldToNewNodeList) {
-        AllEdgesIterator eIter = fromGraph.getAllEdges();
-        while (eIter.next()) {
+    static Graph createSortedGraph(Graph fromGraph, Graph toSortedGraph, final IntIndexedContainer oldToNewNodeList, final IntIndexedContainer newToOldEdgeList) {
```

Comment:
Why is it `oldToNewNodeList` vs. `newToOldEdgeList` ? Should it not be `oldToNewEdgeList` ? As far as I understand this naming it means that I can use an old edge id (from `fromGraph`) as index into this list to retrieve the edge id this edge should have in the new graph (`toSortedGraph`) -> 'old to new'

Reply:
It's `newToOldEdgeList` because we can only append edges to the new graph so we need to iterate through edges in the new order. This list lets us know which old edge we need to get data from for the new edge.  See line 354 where I set the values.

It's backwards  from nodes because `NodeAccess` provides an API to set a node with a given index. As far as I can tell `EdgeAccess` does not have a similar API.

Reply:
Ah yes sorry I should have looked closer. Yes you can only append edges.